File: ApplicationServer/src/ManhwaTrackerApplicationServer/Models/ManhwaList/ManhwaList.cs
using System.ComponentModel.DataAnnotations;

namespace ManhwaTrackerApplicationServer.Models.ManhwaList;

public class ManhwaList
{
    [Key]
    public int Id { get; set; }

    [Required]
    public string? Name { get; set; }
    public string? Description { get; set; }
    public IEnumerable<ManhwaListEntry> ListEntries { get; set; }
    public User.User Owner { get; set; }
}

File: ManhwaTrackerWebsite/manhwa-tracker/models/manhwa-list/manhwa-list.ts
import { User } from "../user/user";
import { ManhwaListEntry } from "./manhwa-list-entry";

export class ManhwaList {
    listEntries: ManhwaListEntry[];
    owner: User;
    id: number;
    name: string;
    description: string;

    constructor(
        owner: User,
        { listEntries }: { listEntries: ManhwaListEntry[] } = {
            listEntries: [],
        },
        id: number,
        name: string,
        description: string
    ) {
        this.owner = owner;
        this.listEntries = listEntries;
        this.id = id;
        this.name = name;
        this.description = description;
    }
}
